### PR TITLE
New feature: unsortable lists

### PR DESCRIPTION
**Pull request by lolmaus**
Please see the readme diff for details.

This PR requires extensive QA.

### CODE REVIEW

**Comment by lolmaus** [comment]
Supercedes #2.

I've built a  demo from this branch and deployed: https://deveo.github.io/ember-drag-sort/

**Comment by PatKoo** [comment]
Checked it out and it works!

File: tests/dummy/app/controllers/index.js [hunk 1]
```diff
@@ -72,14 +104,40 @@ export default Controller.extend({
...
       if (sourceList === targetList && sourceIndex === targetIndex) return
 
       const item = sourceList.objectAt(sourceIndex)
 
       sourceList.removeAt(sourceIndex)
       targetList.insertAt(targetIndex, item)
     },
+
+    determineForeignPosition ({draggedItem, items}) {
+      items = A(items.slice()) // create a copy of the list
```
Comment (vvainio) [inline tests/dummy/app/controllers/index.js]: How about writing this as:
```
return items
  .toArray() // create a copy of the list, convert to Ember.Array if needed: `Ember.Array(items.slice())`
  .addObject(draggedItem)
  .sortBy('name')
  .indexOf(draggedItem)
```
The snippet in the `README.md` could be updated as well.

File: tests/dummy/app/controllers/index.js [hunk 2]
```diff
@@ -72,14 +104,40 @@ export default Controller.extend({
...
+      if (sourceList === targetList && sourceIndex === targetIndex) return
+
+      const unsortableList = this.get('items7')
+
+      let item = sourceList.objectAt(sourceIndex)
+
+      if (sourceList === unsortableList) item = {...item} // shallow clone
+      else sourceList.removeAt(sourceIndex)
+
+      console.log(targetList,  unsortableList, targetList === unsortableList)
```
Comment (vvainio) [inline tests/dummy/app/controllers/index.js]: Could be removed if no longer needed.

File: tests/dummy/app/controllers/index.js (hunk 1 above)
Comment (lolmaus) [inline tests/dummy/app/controllers/index.js]: This has two issues:

1. `.sortBy` is an Ember method, but there's no guarantee that Ember prototype extensions are enabled in the host app.
2. `.addObject` returns the passed object.

Comment (lolmaus) [inline tests/dummy/app/controllers/index.js]: Wait, the latter is not true.

image

This looks like Ember is violating its own conventions. :trollface:

Comment (vvainio) [inline tests/dummy/app/controllers/index.js]: > There's no guarantee that Ember prototype extensions are enabled in the host app.

Good point, didn't think about this 👍

**Comment by lolmaus** [comment]
Published as `1.0.0-beta.0`.

**Pull request merged**